### ISSUE TITLE
bug(mat-button): Focus is imperceptible on mat-button in High Contrast Mode / white-on-black

### ISSUE THREAD

**Issue by JeffKochuk** [angular/components#21363]
#### Reproduction

Steps to reproduce:
1.  In Firefox, set high contrast mode for white-on-black. This can be done in the preferences menu, with preferences -> colors -> Text (set to white), Background (set to black), Override the colors specified by the page with your selections above (Always)
2. Open https://material.angular.io/components/button/overview . 
3. The body tag has the correct cdk a11y classes applied (cdk-high-contrast-active cdk-high-contrast-white-on-black)
4. Tab to the buttons on that page so they receive focus. Instead of the expected white outline, they only receive a slight, grey tint.
 
 
#### Expected Behavior

Focused button should have a 3px solid white outline

#### Actual Behavior

Focused buttons have a 3px solid black outline (which is invisible because of the background)
![Screen Shot 2020-12-15 at 4 52 58 PM](https://user-images.githubusercontent.com/3200118/102277358-26470e80-3ef6-11eb-9e7f-275860e5beea.png)

#### Environment
- Angular:
- CDK/Material:
- Browser(s): Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): macOS

**Comment by JeffKochuk** [comment]
Related issue: https://github.com/angular/components/issues/20820

**Comment by crisbeto** [comment]
Do you happen to have changed your high contrast settings to override the border color? Here's what the docs look like for me on Firefox:

![Button__Angular_Material_–_Mozilla_Firefox_2020-12-16_17-14-45](https://user-images.githubusercontent.com/4450522/102375037-4f34d580-3fc2-11eb-9e06-b73c3f30cc8f.png)

**Comment by JeffKochuk** [comment]
Interesting. I am on macOS and have *only* followed the instructions here:

You can enable Firefox' Windows high contrast mode behavior in MacOS and Linux through a browser preference:

Open about:preferences
Scroll to Language & Appearance
Hit the “Colors…” button
Change the "Override the colors specified by the page with your selections above" select value to “Always”
Ensure “Use system colors“ is unchecked
Ensure that "Background" is set to black and that "Text" is set to white if you are emulating High Contrast Black mode
Hit the "OK" button to save
Restart Firefox





BUT I was able to recreate what YOU see by doing the following:
Follow the above instructions but set text to black, background to white.
in MacOS accessibility settings, check 'Invert Colors'

**Comment by JeffKochuk** [comment]
I'll try to find a windows VM to recreate the setup

**Comment by JeffKochuk** [comment]
Confirmed it looks proper like yours on IE and FF with a dark high-contrast theme on windows and nothing else changed in the settings.

**Comment by JeffKochuk** [comment]
OK this seems to be just an issue with that one particular method of high-contrast-black mode on firefox, and not an issue with the components themselves.
